src/DurationAwareTrait.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace RebelCode\Time;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+
+/**
+ * Common functionality for objects that are aware of a duration.
+ *
+ * @since [*next-version*]
+ */
+trait DurationAwareTrait
+{
+    /**
+     * An integer number of seconds.
+     *
+     * @since [*next-version*]
+     *
+     * @var int
+     */
+    protected $duration;
+
+    /**
+     * Retrieves the timestamp.
+     *
+     * @since [*next-version*]
+     *
+     * @return int An integer number of seconds since unix epoch.
+     */
+    protected function _getDuration()
+    {
+        return $this->duration;
+    }
+
+    /**
+     * Sets the timestamp.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|string|Stringable $duration The number of seconds since unix epoch. Negative numbers are allowed.
+     */
+    protected function _setDuration($duration)
+    {
+        try {
+            $this->duration = $this->_sanitizeTimestamp($duration);
+        } catch (InvalidArgumentException $exception) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not a valid duration'),
+                null,
+                $exception,
+                $duration
+            );
+        }
+    }
+
+    /**
+     * Sanitizes a timestamp to an integer number.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|string|Stringable|null $timestamp The timestamp.
+     *
+     * @return int The sanitized timestamp.
+     */
+    abstract protected function _sanitizeTimestamp($timestamp);
```

Comment:
I think *this* method shouldn't allow `null`: because that indicates an absence of a value, you can't really normalize it to a number. Which is contrary to the setter, which can use `null` to "unset" the value, or simply accept an "empty" value.